app/controllers/TransferSummaryController.scala

```diff
@@ -17,44 +18,65 @@ import scala.concurrent.{ExecutionContext, Future}
 class TransferSummaryController @Inject()(val controllerComponents: SecurityComponents,
                                           val graphqlConfiguration: GraphQLConfiguration,
                                           val keycloakConfiguration: KeycloakConfiguration,
+                                          consignmentService: ConsignmentService,
                                           val consignmentExportService: ConsignmentExportService,
                                           langs: Langs)
                                          (implicit val ec: ExecutionContext) extends ValidatedActions with I18nSupport {
 
   implicit val language: Lang = langs.availables.head
-  val transferSummaryForm: Form[TransferSummaryData] = Form(
+  val transferConfirmationForm: Form[TransferConfirmationData] = Form(
     mapping(
       "openRecords" -> boolean
         .verifying(messagesApi("transferSummary.openRecords.error"), b => b),
       "transferLegalOwnership" -> boolean
         .verifying(messagesApi("transferSummary.transferLegalOwnership.error"), b => b)
-    )(TransferSummaryData.apply)(TransferSummaryData.unapply)
+    )(TransferConfirmationData.apply)(TransferConfirmationData.unapply)
   )
 
-  def transferSummary(consignmentId: UUID): Action[AnyContent] = consignmentExists(consignmentId) { implicit request: Request[AnyContent] =>
-    Ok(views.html.transferSummary(consignmentId, transferSummaryForm))
+  private def getConsignmentSummary(request: Request[AnyContent], consignmentId: UUID)
+                                   (implicit requestHeader: RequestHeader): Future[ConsignmentSummaryData] = {
+    consignmentService.getConsignmentTransferSummary(consignmentId, request.token.bearerAccessToken)
+      .map{summary =>
+        ConsignmentSummaryData(summary.series.get.code,
+          summary.transferringBody.get.name,
+          summary.totalFiles)
+      }
   }
 
-  def transferSummarySubmit(consignmentId: UUID): Action[AnyContent] = secureAction.async { implicit request: Request[AnyContent] =>
-    val errorFunction: Form[TransferSummaryData] => Future[Result] = { formWithErrors: Form[TransferSummaryData] =>
-      Future.successful(BadRequest(views.html.transferSummary(consignmentId, formWithErrors)))
-    }
+  def transferSummary(consignmentId: UUID): Action[AnyContent] = secureAction.async { implicit request: Request[AnyContent] =>
+    getConsignmentSummary(request, consignmentId)
+      .map{ consignmentSummary =>
+        Ok(views.html.transferSummary(consignmentId, consignmentSummary, transferConfirmationForm))
+      }
+  }
 
-    val successFunction: TransferSummaryData => Future[Result] = { _: TransferSummaryData =>
+  def transferSummarySubmit(consignmentId: UUID): Action[AnyContent] =
```

Comment:
This is really cool, took me a second to understand but looks logical and a good way of doing the error handling for the form! :+1:

Reply:
Thanks, but I got the idea from the transfer agreement controller. It's a pretty nifty feature